src/Microsoft.AspNet.Mvc.DataAnnotations/ValidationAttributeAdapterProvider.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.ComponentModel.DataAnnotations;
+using Microsoft.AspNet.Mvc.ModelBinding.Validation;
+using Microsoft.Extensions.Localization;
+
+namespace Microsoft.AspNet.Mvc.DataAnnotations
+{
+    /// <summary>
+    /// Creates a <see cref="IAttributeAdapter"/> for the given attribute.
+    /// </summary>
+    public class ValidationAttributeAdapterProvider
```

Comment:
Why did the design of this class change from that in `DataAnnotationsClientModelValidatorProvider`? Just refactor the code, make this class a singleton, grab it from DI where needed, and call it a day. This approach does not appear to be a step toward any future extensibility story and is no longer replaceable without replacing a great deal of the system.